packages/relay-compiler/codegen/RelayFileWriter.js

```diff
@@ -353,13 +360,39 @@ class RelayFileWriter implements FileWriterInterface {
       return allOutputDirectories;
     });
   }
-}
 
-function md5(x: string): string {
-  return crypto
-    .createHash('md5')
-    .update(x, 'utf8')
-    .digest('hex');
+  /**
+   * Find all *.queryMap.json and write it into a single file.
+   * @param allOutputDirectories
+   */
+  writeCompleteQueryMap(
+    allOutputDirectories: Map<string, CodegenDirectory>,
+  ): void {
+    const queryMapFilePath = `${this._config.baseDir}/queryMap.json`;
```

Comment:
I’m wondering if it would make sense to take an optional path value for the `--persist` option to change this location?

Reply:
Would be nice, in our app we have multiple compilations going on... so I guess without that, it would overwrite existing compilation.

Reply:
@edvinerikson each one of your compilation should have a `--src` specified, which means a separate `queryMap.json` file will be created for each src. Is that your use case? If not can you please provide more details on what you mean by multiple compilations?

Reply:
I am executing them like this
```
    "compile-gql-documents:desktop": "relay-compiler --schema ./node_modules/@leogears/leo-gateway-schema/schema.json --src ../../ --include src/common/** --include src/web/common/** --include src/web/desktop/**",
    "compile-gql-documents:mobile": "relay-compiler --schema ./node_modules/@leogears/leo-gateway-schema/schema.json --src ../../ --include src/common/** --include src/web/common/** --include src/web/mobile/**",
```

Reply:
Gotcha. You are right in your case the query map files will override each other. I will add an option to specify an output location. Thanks for the feedback!

Reply:
>I’m wondering if it would make sense to take an optional path value for the --persist option to change this location?
Done. @edvinerikson @alloy